### PR TITLE
cmd/snap,client,daemon: support set/unset of store front

### PR DESCRIPTION
**Pull request by atomatt**
Simple CLI command to set or revert the current store API from an ack'd store assertion.

### CODE REVIEW

File: daemon/api.go [hunk 1]
```diff
@@ -2688,3 +2701,86 @@ func postApps(c *Command, r *http.Request, user *auth.UserState) Response {
...
+
+	// URL is optional, but it makes no sense to set a store with no URL.
+	if storeURL == nil {
+		return BadRequest("store assertion with store %q has no URL", storeData.Store)
+	}
+
+	// Replace active store and update state.
+	st.Lock()
+	defer st.Unlock()
+	err = storestate.SetBaseURL(st, storeURL)
```
Comment (zyga) [inline daemon/api.go]: Should we ensure that there are no changes in flight?

File: daemon/api.go [hunk 2]
```diff
@@ -2688,3 +2701,86 @@ func postApps(c *Command, r *http.Request, user *auth.UserState) Response {
...
+	errResponse := checkRootUser(r, "revert store")
+	if errResponse != nil {
+		return errResponse
+	}
+
+	// Replace active store and update state.
+	st := c.d.overlord.State()
+	st.Lock()
+	defer st.Unlock()
+	err := storestate.SetBaseURL(st, nil)
```
Comment (zyga) [inline daemon/api.go]: Ditto

File: daemon/api.go (hunk 1 above)
Comment (atomatt) [reply daemon/api.go]: Good question! But I think we're ok - there's a single func, storestate.SetBaseURL, that replaces the current store and updates state, and that func requires a state lock to succeed.

**Comment by atomatt** [comment]
Note: if this is basically ok, we almost certainly want to change the command name as "store" is a bit ambiguous and could easily be interpreted as a way to store something.

**Comment by chipaca** [comment]
@niemeyer asking for your eyes here because it involves a toplevel snap verb

**Comment by niemeyer** [comment]
How about:
```
$ snap through <name>
Documentation along the lines of:
```
The through command changes the configuration of snapd so it
communicates via the named proxy store. An assertion that defines
the details for the proxy store must have been previously loaded
into the system.
```
If the command is executed without arguments, it prints the name of
a previously configured store, or "unset" if nothing was configured.
We can also support a flag to unset it explicitly:
```
$ snap through --disable
A new "through" entry is added into /v2/system-info (if non-empty), and
this same endpoint gets a new POST action that may change the setting.
```
If the parameter is provided on POST but the value is empty, we can unset it.

**Comment by pedronis** [comment]
this will be done very differently

**Pull request closed without merging**